caffe2/operators/reshape_op.h

```diff
@@ -46,14 +46,20 @@ class ReshapeOp : public Operator<Context> {
       auto& shape = Input(1);
       CAFFE_ENFORCE(shape.dim() == 1, "Shape should be 1-D");
 
-      const T* shape_data = shape.template data<T>();
-
-      // Bit awkward, but needed so works on both CPU and CUDA contexts
-      std::vector<T> tmpv(shape.numel());
-      if (shape.numel() > 0) {
-        context_.CopyBytesToCPU(
-            shape.numel() * sizeof(T), shape_data, &tmpv[0]);
-        actual_new_shape.assign(tmpv.begin(), tmpv.begin() + shape.numel());
+      if (shape.dtype() == TypeMeta::Make<int64_t>() ||
```

Comment:
Why do we need to change the implementation of Reshape in Caffe2?

Reply:
Well, this is not changing but rather adding one more use case when `Input(1)` appears to be float, thus it fails here. It happens with ONNX zoo models of opset 5+. So, as you can see, we have if/else fork here. However, I might be wrong and should reject such cases. Indeed, when I get `shape=[1.4678, 0.43289]` what exactly this means?